### PR TITLE
Add match field in PolicyTemplateDetails pages for gatekeeper

### PR DESCRIPTION
**Pull request by yiraeChristineKim**
add the "matches" row with "pills" for each match entry.
Signed-off-by: yiraeChristineKim <yikim@redhat.com>

### CODE REVIEW

**Comment by KevinFCormier** [comment]
@yiraeChristineKim I see that the Konflux job has now passed, but for future reference, the way to rerun it is to comment `/retest`

**Comment by yiraeChristineKim** [comment]
> @yiraeChristineKim I see that the Konflux job has now passed, but for future reference, the way to rerun it is to comment `/retest`

Thanks Kevin!

File: frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetails.tsx [hunk 1]
```diff
@@ -265,3 +283,51 @@ export function PolicyTemplateDetails() {
     </div>
   )
 }
+
+const matchesBadges = (kinds: IKinds[]): ReactNode => {
+  const badgeStyle: CSSProperties = {
```
Comment (mprahl) [inline frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetails.tsx:288]: Could you use the Patternfly badge instead for consistent UX?

File: frontend/src/routes/Governance/discovered/useFetchPolicies.tsx [hunk 2]
```diff
@@ -133,6 +133,10 @@ export function useFetchPolicies(policyName?: string, policyKind?: string, apiGr
       searchDataItems = searchDataItems.concat(result?.items || [])
     })
 
+    if (searchDataItems.length == 0 && !searchErr && !searchLoading) {
```
Comment (yiraeChristineKim) [inline frontend/src/routes/Governance/discovered/useFetchPolicies.tsx:136]: Critical bug

**Comment by mprahl** [comment]
@yiraeChristineKim could you fix the typo in your commit message `filed` => `field`?

File: frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetails.tsx [hunk 3]
```diff
@@ -265,3 +283,41 @@ export function PolicyTemplateDetails() {
     </div>
   )
 }
+
+const matchesBadges = (kinds: IKinds[]): ReactNode => {
+  return (
+    <MuiGrid
+      container
+      direction="column"
+      style={{ gap: 8 }}
```
[the remark this answers is not in the document]
Comment (yiraeChristineKim) [reply frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetails.tsx:292]: okie!!

**Review by mprahl — APPROVED** [review]

**Pull request merged**